src/components/CategoryFilter.js

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { v4 as uuidv4 } from 'uuid';
+
+const CATEGORIES = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
+
+const CategoryFilter = (props) => {
+  const { filter, handleFilterChange } = props;
+  const FILTERS = ['All', ...CATEGORIES];
+
+  return (
+    <select name="filter" value={filter} onChange={handleFilterChange}>
+      {FILTERS.map((category) => (
+        <option key={uuidv4()} value={category}>
+          {category}
+        </option>
+      ))}
+    </select>
+  );
+};
+
+CategoryFilter.propTypes = { filter: PropTypes.string.isRequired };
+CategoryFilter.propTypes = { handleFilterChange: PropTypes.func.isRequired };
```

Comment:
- [ ] [MANDATORY] Reassignment is not actually the correct way to implement multiple props validation.  this means that whichever the value assigned last will replace everything and be the only value for that variable. in this   case, `CategoryFilter.propTypes` will be equal to `{ handleFilterChange: PropTypes.func.isRequired }` after this file is exuted. that means `filter is not defined in there anymore`. Here is the correct way to do this: 
```Javascript
    CategoryFilter.propTypes = { 
       filter: PropTypes.string.isRequired,
       handleFilterChange: PropTypes.func.isRequired, 
    };
```